ibis/__init__.py

```diff
@@ -51,15 +53,29 @@
 __version__ = get_versions()['version']
 del get_versions
 
-for entry_point in pkg_resources.iter_entry_points(
-    group='ibis.backends', name=None
-):
-    try:
-        backend_module = entry_point.resolve()
-    except ImportError:
-        pass
-    else:
-        backend = backend_module.Backend()
-        setattr(ibis, entry_point.name, backend)
-        with ibis.config.config_prefix(entry_point.name):
-            backend.register_options()
+
+def __getattr__(name: str):
+    entry_points = list(
+        pkg_resources.iter_entry_points(group='ibis.backends', name=name)
+    )
+    if len(entry_points) == 0:
+        raise AttributeError(
+            f"module 'ibis' has no attribute '{name}'. "
```

Comment:
are these 2 tested? why are we warning on multiple entry points? (and not raising)